codex-cli/src/utils/check-in-vcs.ts

```diff
@@ -29,3 +29,45 @@ export function checkInGit(workdir: string): boolean {
     return false;
   }
 }
+
+/**
+ * Returns true if the given directory is part of a Mercurial (hg) repository.
+ *
+ * Uses the canonical Mercurial command `hg root` which exits with status 0
+ * when executed anywhere inside a working tree (including the repo root) and
+ * exits with a non-zero status otherwise. We only rely on the exit code.
+ */
+export function checkInHg(workdir: string): boolean {
+  try {
+    execSync("hg root", { cwd: workdir, stdio: "ignore" });
+    return true;
+  } catch {
+    return false;
+  }
+}
+/**
+ * Returns true if the given directory is part of a Sapling (sl) repository.
+ *
+ * Uses the Sapling command `sl root` which exits with status 0 when executed
+ * anywhere inside a working tree and exits with a non-zero status otherwise.
+ */
+export function checkInSapling(workdir: string): boolean {
+  try {
+    execSync("sl root", { cwd: workdir, stdio: "ignore" });
```

Comment:
I'm not sure what will happen for folks who have https://github.com/mtoyoda/sl installed.

Reply:
I assumed it will fail with non 0 exit code, resulting in returning false (same if none of the `sl`s is not installed)